nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -307,6 +328,20 @@ private boolean commit(final ProcessContext context, final ProcessSession sessio
         return willCommit;
     }
 
+    private Map<String, String> writeObjectTags(AmazonS3 client, S3VersionSummary versionSummary) {
+        final GetObjectTaggingResult taggingResult = client.getObjectTagging(new GetObjectTaggingRequest(versionSummary.getBucketName(), versionSummary.getKey()));
```

Comment:
Can't have a deeper look right now but is the retrieval of tags requiring an additional call to the API or is the info already retrieved anyway? If this line causes an additional call to the API, I'd recommend to change the default value of the property to "false" so that existing/running workflows are not impacted by this change. Thoughts?

Reply:
Yep. It'll make additional API call but I don't think it will have any negative impact on the existing flows? Do you sense any cases where this might break? This simply calls S3 service to see if this key has any tags associated, if it has, it will add to the flowfile attributes.

Reply:
I just imagine the case where a user is listing a lot of objects in S3 and this could cause a large amount of additional calls. I agree that the overhead should be fairly limited. Just wondering if we want this new behavior to be enabled by default. I honestly don't have any strong opinion on that, just wanted to mention it. Do you have an opinion on that @jvwing ?

Reply:
I agree with @pvillard31 that it should be off by default.  From comments on the users/developer email lists, I understand ListS3 is used to process very large lists of objects, easily 10,000+ on a regular basis.  Even if the additional API calls are quick, it will add up to be a lot of API calls.

Unfortunately, it does not look like the [S3ObjectSummary](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/S3ObjectSummary.html) returned by the listing contains any hints on the number of tags present, if any.

Reply:
I agree. I'll change the default value to `false`.